### PR TITLE
chore: Yt validate with state

### PR DESCRIPTION
**Pull request by dipeshkemuri**

### CODE REVIEW

File: src/components/UpcomingEvents.vue [hunk 1]
```diff
@@ -63,8 +65,27 @@ export default {
...
+      const ytUIDs = JSON.parse(event.target?.firstElementChild?.getAttribute('data-uid'));
+      let validUID = '';
+      let i = 0;
+
+      while (i < ytUIDs.length) {
+        await validateYTVideo(ytUIDs[i])
+          .then(res=> {
+            if (res.isValid) {
+              validUID = res.uid;
+              event.target?.firstElementChild?.setAttribute('src', 'https://www.youtube.com/embed/' + validUID + '?autoplay=1&mute=1&controls=0');
```
Comment (akshay-kh) [inline src/components/UpcomingEvents.vue:81]: use template literal

File: src/components/EventsTemplate.vue [hunk 2]
```diff
@@ -159,8 +163,24 @@ const hitSearchAPI = () => debounce(()=>{
...
+  const ytUIDs = JSON.parse(event.target?.firstElementChild?.getAttribute('data-uid'));
+  let validUID = '';
+  let i = 0;
+
+  while (i < ytUIDs.length) {
+    await validateYTVideo(ytUIDs[i])
+      .then(res=> {
+        if (res.isValid) {
+          validUID = res.uid;
+          event.target?.firstElementChild?.setAttribute('src', 'https://www.youtube.com/embed/' + validUID + '?autoplay=1&mute=1&controls=0');
```
Comment (akshay-kh) [inline src/components/EventsTemplate.vue:176]: use template literal

File: src/components/UpcomingEvents.vue [hunk 3]
```diff
@@ -46,7 +46,9 @@
   
 <script>
 import { ArrowRightIcon } from '@heroicons/vue/20/solid'
+
```
Comment (akshay-kh) [inline src/components/UpcomingEvents.vue:49]: maybe this change is not required ?

**Review by akshay-kh — CHANGES_REQUESTED** [review]

File: src/components/EventsTemplate.vue (hunk 2 above)
Comment (dipeshkemuri) [reply src/components/EventsTemplate.vue]: Applied

File: src/components/UpcomingEvents.vue (hunk 1 above)
Comment (dipeshkemuri) [reply src/components/UpcomingEvents.vue]: Applied

**Review by akshay-kh — APPROVED** [review]

**Pull request merged**